frontend/packages/yki/src/utils/examSession.ts

```diff
@@ -0,0 +1,249 @@
+    return (
+      examSession.post_admission_active &&
+      examSession.post_admission_start_date &&
+      examSession.post_admission_end_date &&
+      examSession.post_admission_quota
+    );
+  }
+
+  private static getPostAdmissionAvailablePlaces(examSession: ExamSession) {
+    if (
+      ExamSessionUtils.isPostAdmissionAvailable(examSession) &&
+      !ExamSessionUtils.hasPostAdmissionEnded(examSession, dayjs())
+    ) {
+      return examSession.post_admission_quota - examSession.pa_participants;
+    }
+
+    return 0;
+  }
+
+  static getAvailablePlaces(examSession: ExamSession) {
+    return !ExamSessionUtils.hasRegistrationEnded(examSession, dayjs())
+      ? ExamSessionUtils.getRegistrationAvailablePlaces(examSession)
+      : ExamSessionUtils.getPostAdmissionAvailablePlaces(examSession);
+  }
+
+  static hasRoom(examSession: ExamSession) {
+    return ExamSessionUtils.getAvailablePlaces(examSession) > 0;
+  }
+
+  private static compareExamSessionsByLang(es1: ExamSession, es2: ExamSession) {
+    // TODO: write proper language comparator / re-order ExamLanguage enums
+    if (es1.language_code < es2.language_code) {
+      return -1;
+    } else if (es1.language_code > es2.language_code) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  private static compareExamSessionsByRoom(es1: ExamSession, es2: ExamSession) {
+    const hasRoom1 = ExamSessionUtils.hasRoom(es1);
+    const hasRoom2 = ExamSessionUtils.hasRoom(es2);
+
+    if (hasRoom1 && !hasRoom2) {
+      return -1;
+    } else if (!hasRoom1 && hasRoom2) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  private static compareExamSessionsByQueueFullness(
+    es1: ExamSession,
+    es2: ExamSession
+  ) {
+    if (!es1.queue_full && es2.queue_full) {
+      return -1;
+    } else if (es1.queue_full && !es2.queue_full) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  private static compareExamSessionsByDate(es1: ExamSession, es2: ExamSession) {
+    if (es1.session_date.isBefore(es2.session_date)) {
+      return -1;
+    } else if (es1.session_date.isAfter(es2.session_date)) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  private static compareExamSessionsByRegistrationEnding(
+    es1: ExamSession,
+    es2: ExamSession
+  ) {
+    const now = dayjs();
+    const registrationEnded1 = ExamSessionUtils.hasRegistrationEnded(es1, now);
+    const registrationEnded2 = ExamSessionUtils.hasRegistrationEnded(es2, now);
+
+    if (!registrationEnded1 && registrationEnded2) {
+      return -1;
+    } else if (registrationEnded1 && !registrationEnded2) {
+      return 1;
+    }
+
+    return 0;
+  }
+
+  static compareExamSessions(es1: ExamSession, es2: ExamSession) {
+    // Prioritised ordering of comparators
+    const comparatorFns = [
+      ExamSessionUtils.compareExamSessionsByLang,
+      ExamSessionUtils.compareExamSessionsByRoom,
+      ExamSessionUtils.compareExamSessionsByQueueFullness,
+      ExamSessionUtils.compareExamSessionsByDate,
```

Comment:
Pitäisköhän itse asiassa vaihtaa nämä 2 päittäin?

Reply:
Vaikea sanoa.. mun mielestä voisi olla ihan loogista myös että tutkinnot listattaisiin ensiksi tutkintopäivän perusteella. Varmasti sekä hyötyä että haittaa nykyisestä järjestyksestä. Ehkä voidaan antaa olla as-is siihen asti että saadaan OPH:lta asiaan jokin mielipide.